### DISCUSSION TITLE
Advice on building a 4G enabled Frigate extension that runs on a vehicle

### DISCUSSION THREAD

**Discussion by mbalsam** [blakeblackshear/frigate#9360] [Ask A Question] [answered]
I’m looking for some advice on how to implement a few Frigate custom features. 

I’m building a system that will: 

1)	Have several cameras mounted on the outside of a truck.  The truck will have a 4G router to communicate with the cloud.
2)	My goal is to conserve 4G bandwidth as no one will be watching 99% of the time.
3)	No viewing of data will occur locally on the truck.  Users will interact with a cloud portal we’ve developed. 
4)	If an event occurs, the system will locally record 30 second MPEG.    When this recording is complete,  a thumbnail will be uploaded to my storage.  (amazon s3 bucket)
5)	The users will see the uploaded thumbnails.  If the users clicks it 
a.	A MQTT message will be sent down to Frigate containing the video id.
b.	That video will be uploaded to S3.  
c.	When it arrives, the video will start playing in the customer portal
6)	Another option will allow the user to request a live stream.  If they do so: 
a.	A MQTT message will be sent down to Frigate. 
b.	A program will listen for these events and when found will initiate a RSTP stream with the camera using go2rtc.
c.	The destination address for the stream will be some type of RSTP proxy server or rebroadcast server.  (wowza?)
d.	The portal will be directed to open a WebRTC  connection to the proxy server on (wowza?)

I understand that by default this functionality is not out of the box with Frigate but some parts do already exist. E.g.  MQTT upload of thumbnails. Etc.

What im looking for is:  Any existing open-source projects that implement any or all of this. If not exist, what will be the best way to implement this?

Does an opensource project exist that runs out side of home assistant that might better match my requirements?

**Comment by mbalsam** [comment, marked as answer]
I'm building this now.  will hopefully release it on github soon.

**Reply by YBonline** [reply]
That sounds awesome, make sure to post when you get something usable, I'd like to try it out at some point!
